Supervisor feedback discussion (#98)
chhoumann/p8: report/src/discussion/discussion.tex

\chapter{Discussion}\label{chap:discussion}
This project involved two main phases: initially, we designed LoFi and HiFi prototypes for a tablet-based meeting room management system, which were well-received by the stakeholder, Mariendal IT.
By choosing a user-centered prototype evaluation, described in Section \ref{sec:prototype_evaluation}, we examined the stakeholder's problems from their point of view.

> One benefit of using a user-centered prototype evaluation is that the users can provide insights into the inner workings of the application domain that would otherwise remain undiscovered.

Reviewer: Yes! But shouldn't this have been written earlier in the report?

Author: Maybe, perhaps. I do not think it would benefit the report mentioning it earlier. Besides, at that point in the process we did not know it would be problematic. I think that the discussion is the perfect place to discuss what we might have gained by using that specific method. We already have academic reasons for using the method, but this clarifies the benefits for this project.

Reviewer: OK. Fine reason. Thanks.

We found that this was indeed the case, as the focus of the project pivoted after this evaluation. 

> Before the evaluation, the focus was on discovering what system-integrations would be needed for a solution to the stakeholder's meeting management problems.

Reviewer: ```suggestion
Before the evaluation, the focus was on discovering what system integrations would be needed for a solution to the stakeholder's meeting management problems.
```

After the evaluation, the project took a turn towards discovering solutions that could automize aspects of a problem solution, leading to the problem formulation in Section \ref{sec:problem_statement}.

> While the results of this evaluation were fruitful, the approach turned out to be very time-consuming due to the amount of work needed to construct HiFi prototypes.

Reviewer: ```suggestion
While the results of this evaluation were fruitful, the approach turned out to be time-consuming due to the amount of work needed to construct HiFi prototypes.
```

In Chapter \ref{chap:presence_intro}, we presented five metrics that would be used to evaluate a selection of technologies for presence detection.
Based on our research, we determined that BLE satisfied most of these metrics in relation to our stakeholders problem.
Using BLE allowed us to consider several techniques for presence detection purposes of which we considered lateration and fingerprinting as primary candidates.
Due to the need for non-noisy environments for lateration to work, we determined that the most appropriate technique for presence detection would be fingerprinting along with kNN for classification.

By choosing BLE and fingerprinting, we could utilize established protocols and beacons to compute locations after a fingerprinting map was created. 

> However, creating this map can be time-consuming, as our method relies on signals with little fluctuation, as high fluctuations causes a high standard deviation. 

Reviewer: ```suggestion
However, creating this map can be time-consuming, as our method relies on signals with little fluctuation, and large fluctuations lead to a high standard deviation. 
```

> Therefore, if one wants to create a map over a large area where large fluctuations occur, the process can become timewise infeasible.

Reviewer: ```suggestion
Therefore, if one wants to create a map over a wide area where large fluctuations occur, the process can become timewise infeasible.
```

> We then implemented a solution utilizing BLE beacons and fingerprinting, and presented this it and results of the system in Chapter \ref{chap:system_design}.

Reviewer: ```suggestion
We then implemented a solution utilizing BLE beacons and fingerprinting, and presented this, as well as results of the system, in Chapter~\ref{chap:system_design}.
```

We found that utilizing four beacons to outline a room and a map generated by traversing the room in a grid pattern of $1m \times 1m$ we were able to achieve significant results.
Depending on the chosen \textit{k} for nearest neighbor search, we were able to achieve between $95.24$ and $97.62$ percent accuracy for the inside case and between $45.33$ and $78.67$ percent accuracy for the outside scenario. 

As mentioned in Section \ref{sec:ChoiceOfTechnique}, the goal of the project was not to achieve high accuracy and precision on the exact positioning of people in a room. 
The goal was to investigate whether we could detect the presence of people in a meeting with reasonable certainty.

> The results show that while we can detect with high precision that the room is in use, the precision of the classification of when the room is unused is more dubious. 

Reviewer: ```suggestion
The results show that while we can detect with high precision that the room is in use, the precision of the classification for when the room is unused is more uncertain. 
```

However, for the purpose of this system, this is not overly problematic. 

> When the occupants exits and leaves the room after a meeting, the between the room and the phone increases, and the classification naturally skews towards \textit{outside}.

Reviewer: ```suggestion
When the occupants exits and leaves the room after a meeting, the distance between the room and the receiving device increases, and the classification naturally skews towards \textit{outside}.
```

Furthermore, these findings align with what similar research has concluded; that using kNN is sufficiently accurate, and that complex machine learning is not necessary to obtain useful results.\cite{ble_kneares_neural}

It is worth noticing that for measuring points \textit{B} and \textit{J} in Figure \ref{fig:experiment_room}, a large uncertainty in classification is observed.
We suspect that this is a consequence of the glass pane next to the door having a much lower absorbation of radio frequencies than the door and the wall next to it.
This low absorbation results in the two data points next to the pane having similar distances to both map values labelled as \textit{inside} and some as \textit{outside}, therefore giving uncertain classifications.

A limitation of the presented approach is that it requires employees to carry a device capable of running the application in the background.
Therefore, the approach can be unsuitable for organizations that do not provide their employees with such devices.
It is a tradeoff between convenience and accuracy.

> Proximity detection, for example, would have been more convenient, since it would only require employees to carry a Bluetooth-enabled device (such as a mobile phone or Bluetooth tag). However, this method is far less accurate, as discussed in Section~\ref{sec:proximity_detection}.

Reviewer: ```suggestion
In this case, proximity detection could have been more convenient, as it would only require employees to carry a Bluetooth-enabled device, such as a mobile phone or Bluetooth tag.
However, this method is far less accurate, as discussed in Section~\ref{sec:proximity_detection}.
```
